influxdb/resource_user.go

```diff
@@ -167,7 +167,7 @@ func readGrants(d *schema.ResourceData, meta interface{}) error {
 		if result[1].(string) != "NO PRIVILEGES" {
 			var grant = map[string]string{
 				"database":  result[0].(string),
-				"privilege": strings.ToLower(result[1].(string)),
+				"privilege": strings.Replace(strings.ToLower(result[1].(string)), "all privileges", "all", 1),
```

Comment:
`strings.ToLower` here is lowercasing all the grant strings and storing them as lowercase in the state file. Grants like these:

```hcl
grant {
  database  = "${influxdb_database.green.name}"
  privilege = "ALL"
}

grant {
  database  = "${influxdb_database.blue.name}"
  privilege = "WRITE"
}
```

Result in a reoccuring diff:

```
  ~ influxdb_user.test
      grant.2079896074.database:  "" => "terraform-blue"
      grant.2079896074.privilege: "" => "WRITE"
      grant.2960020804.database:  "terraform-blue" => ""
      grant.2960020804.privilege: "write" => ""
      grant.3779313284.database:  "" => "terraform-green"
      grant.3779313284.privilege: "" => "ALL"
      grant.3971511288.database:  "terraform-green" => ""
      grant.3971511288.privilege: "all" => ""
```

I looked at the InfluxDB docs, and they suggest these values are always uppercase, though I don't recall seeing something explicitly saying they will always be upper case 🤔 

The example grants in the [documentation for the resource](https://www.terraform.io/docs/providers/influxdb/r/user.html#grant) use upper case, so I recommend we use a validation function for the input to ensure uppercase is used. Here's an example of a validation func in use:

- https://github.com/terraform-providers/terraform-provider-aws/blob/879ea1bed3ff54c48ffbb64bbfd9244bc7b98720/aws/resource_aws_iam_role.go#L65-L77